### ISSUE TITLE
Updated Version of Semantic Segmentation Inference Engine Deployment for TensorRT 8 and higher

### ISSUE THREAD

**Issue by ashutoshraman** [NVIDIA/TensorRT#4445]
Hi,
I am trying to implement the script for the Jupyter notebook for tensorRT semantic segmentation using a .trt engine I built for the NVDA Jetson Nano Orin using regular command line directions. I am now trying to write an inference script that deserializes and uses my trt inference engine in segmenting my test dataset images. However, it seems the most recent update of this script was 4 years ago, using an older version of TensorRT. I am using TensorRT version 8.3 I believe, as part of my Jetpack SDK version 6.2 for my NVDA Jetson Orin, and this same script, adapted for my dataset, does not work. Is there a new or updated version of this script, or a tutorial somewhere on how to do all these steps with the new tensorrt, so that I can deploy my custom (from-scratch) deep segmentation model? I am a beginner to embedded programming and quantized deployment, so I would appreciate any help. If no tutorial or code exists, could I possibly show my current code and get input on how to adapt it for trt version 8.3 or to see where I am messing up? Thank you

**Comment by poweiw** [comment]
Hello @ashutoshraman are you referring to this example? https://github.com/NVIDIA/TensorRT/blob/release/10.10/quickstart/SemanticSegmentation/tutorial-runtime.ipynb

If so, this example has been tested with the latest TRT version `10.10` so I would recommend trying that first

**Comment by ashutoshraman** [comment]
Hi,
I was mistaken, I am actually using TensorRT version 10.3. I was using this example you linked but for an older version, which was linked directly on the Nvidia site. I am trying the 10.3 version example out now.

I am encountering issues in that while it runs and completes properly, my output is very uniform, it appears that masks are all the exact same with minor variatons for every image. I am inputting (1,1,1024,2048) images and expecting an output array of size (1,3,1,2048), since I am doing column based classification. The model infers just fine on my computer using torch commands, but using TensorRT on my Jetson Orin Nano, it produces almost complete uniformity.

 In the images, which are vertically stacked output arrays that form an en face image, the TRT output is uniform and incorrect columns (implying each prediction row is basically the same), while the desktop inference is more heterogeneous and what is expected. I am happy to link my code if this would be helpful. I was getting this same issue when I was using the older example for the Segmentation as well.

**Comment by poweiw** [comment]
> I am happy to link my code if this would be helpful

@ashutoshraman thanks for the response! So it seems like you have your modified version of the notebook? If so then a link to your code would be very useful 👍 I would also encourage to try with the latest TRT version

**Comment by ashutoshraman** [comment]
Ok yes here is a link to my inference code using the trt engine I created based off an onnx conversion from a pth weights file. I am using a Jetson Orin Nano Dev Kit

[share_trt_inference.txt](https://github.com/user-attachments/files/20510787/share_trt_inference.txt)

**Comment by ashutoshraman** [comment]
Hello,
I was wondering if you have gotten the chance to review this file. I also have been noticing, I created an engine and specified precision of fp16 but when i input fp16 numpy-based images to the inference engine, I get output of torch.float32, so I wonder if this is related to the issue. I also ran the onnx file as part of an inference script and the output was the correct output with heterogeneous array predictions as hoped for. So I am fairly confident there is an issue with the conversion to tensorrt or tensorrt engine usage portions.

Could you please provide your thoughts on this script and help me understand what I am doing wrong?

**Comment by poweiw** [comment]
Sorry for the late reply @ashutoshraman . Will take a look this by end of this week. 
Also I don't see an engine file provided. Would it be possible to provide it for better verification?

**Comment by ashutoshraman** [comment]
Ok thank you. It wont let me attach .engine files, how can i provide this to you?

Here is a google drive link, I can leave it up for a week. 
https://drive.google.com/drive/folders/16b2XBGQupsJDR4akAyzKSPNx0mdnlhyR?usp=sharing

**Comment by ashutoshraman** [comment]
Hi,
I figured it out, it was not an issue with TRT, but rather ONNX and the way I was exporting or evaluating my model. Everything works as desired now. Thank you and sorry to bother

**Comment by ashutoshraman** [comment]
Hi,
I figured it out, it was not an issue with TRT, but rather with ONNX exporting and some preprocessing. It works now

**Comment by poweiw** [comment]
Happy to hear! @ashutoshraman
